### DISCUSSION TITLE
How to give more training time to generator and less to disciminator in GAN-like model

### DISCUSSION THREAD

**Discussion by MarcoAlbertoGrimaldi** [Lightning-AI/pytorch-lightning#6085] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi,
as title I was wandering how I can train more one sub-model than another.
Like 5 epochs on generator and 1 on discriminator
My code is:

`     def configure_optimizers(self):
        #stuff
        return [optimizer_G, optimizer_D], []

    def training_step(self, train_batch, batch_idx, optimizer_idx):

        grid_real, real = train_batch
        grid_fake = self.generator(real)

        valid = torch.ones(self.patch_dim, requires_grad=False, device=self.device).type_as(real)

        if optimizer_idx == 0:
            sample_img = grid_fake[0]
            self.logger.experiment.add_image('generated_images', sample_img, 0)
            g_loss = self.gan_loss(self.discriminator(grid_fake, grid_real),
                                   valid) + self.hparams.lambda_multiplier * self.pixel_loss(grid_fake, grid_real)
            self.log('generator_loss', g_loss)
            return g_loss

        if optimizer_idx == 1:
            invalid = torch.zeros(self.patch_dim, requires_grad=False, device=self.device).type_as(real)
            predict_real = self.discriminator(grid_real, real)
            predict_fake = self.discriminator(grid_fake.detach(), real)
            real_loss = self.gan_loss(predict_real, valid) * 0.5
            fake_loss = self.gan_loss(predict_fake, invalid) * 0.5
            d_loss = real_loss + fake_loss
            self.log('discriminator_loss', d_loss, sync_dist=True)
            return d_loss`

**Comment by sakvaua** [comment]
if self.current_epoch%5:
#do discriminator
